src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -298,8 +299,26 @@ static async Task WaitForHighPriorityTasksAsync(CancellationToken cancellationTo
                         return;
                     }
 
-                    // Wait for all the high priority tasks, ignoring all exceptions from it.
-                    await Task.WhenAll(highPriorityTasksToAwait).WithCancellation(cancellationToken).NoThrowAwaitable(false);
+                    // Wait for all the high priority tasks, ignoring all exceptions from it. Loop directly to avoid
+                    // expensive allocations in Task.WhenAll.
+                    foreach (var task in highPriorityTasksToAwait)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        await WaitForHighPriorityTaskAsync(task, cancellationToken).ConfigureAwait(false);
+                    }
+                }
+
+                static async Task WaitForHighPriorityTaskAsync(Task task, CancellationToken cancellationToken)
```

Comment:
nit: feel free to inline this.  I don't know if it is super relevant to have as a helper method.

Reply:
Yeah, I'd suggest just doing it inline, e.g.
```C#
foreach (var task in highPriorityTasksToAwait)
{
    cancellationToken.ThrowIfCancellationRequested();
    if (task.IsCompleted)
    {
        // Make sure to yield so continuations of 'task' can make progress.
        await Task.Yield().ConfigureAwait(false);
    }
    else
    {
        await task.WithCancellation(cancellationToken).NoThrowAwaitableInternal(false);
    }
}
```